torch/_refs/__init__.py

```diff
@@ -2417,6 +2418,36 @@ def uniform(
     return prims.uniform(shape, low=low, high=high, dtype=dtype, device=device)
 
 
+def masked_fill(a: TensorLikeType, mask: TensorLikeType, value: TensorOrNumberLikeType):
+    python_type = utils.dtype_to_type(a.dtype)
+    if isinstance(value, Number):
```

Comment:
Let's file an issue for this unconditional cast -- fill_ has similar behavior and I think it's a bug -- then let's comment the issue here